client/components/notice/index.jsx

```diff
@@ -108,6 +109,10 @@ export default React.createClass( {
 			noticeClass = classnames( noticeClass, 'is-compact' );
 		}
 
+		if ( this.props.isSmallViewport ) {
+			noticeClass = classnames( noticeClass, 'is-small-viewport' );
```

Comment:
This is more a general refactoring point for the whole component, but it's more concise, clean, and performant to pass optional modifiers as the second object argument:

```js
const noticeClass = classnames( 'notice', this.props.status, this.props.className, {
	'is-compact': this.props.isCompact,
	'is-small-viewport': this.props.isSmallViewport,
	'is-dismissable': this.props.showDismiss
} );
```